src/track.js

```diff
@@ -24,13 +24,18 @@ function extractGPXTracks(gpx) {
         let name = trk.name && trk.name.length > 0 ? trk.name[0] : 'untitled';
 
         trk.trkseg.forEach(trkseg => {
-            let points = trkseg.trkpt.map(trkpt => ({
-                lat: parseFloat(trkpt.$.lat),
-                lng: parseFloat(trkpt.$.lon),
-                // These are available to us, but are currently unused
-                // elev: parseFloat(trkpt.ele) || 0,
-                // time: new Date(trkpt.time || '0')
-            }));
+            let points = trkseg.trkpt.reduce(function(result, trkpt) {
+                if (typeof trkpt.$ !== 'undefined') {
```

Comment:
Would it be better to check if `$.lat` and `$.lon` are defined, in case other attributes are present?

Reply:
Thanks, updated in https://github.com/erik/derive/pull/27/commits/6b3f2e191c2b3f3587e716022829bec0ab405755.

The inital check for `typeof trkpt.$ !== 'undefined'` is still needed to prevent `TypeError: Cannot read property 'lat' of undefined`.